packages/edit-site/src/components/sidebar-navigation-screen-pages/index.js

```diff
@@ -4,61 +4,150 @@
+		homeTemplate,
+	] );
+
+	const { frontPage, postsPage } = useSelect( ( select ) => {
+		const { getEntityRecord } = select( coreStore );
+
+		const siteSettings = getEntityRecord( 'root', 'site' );
+		return {
+			frontPage: siteSettings?.page_on_front,
+			postsPage: siteSettings?.page_for_posts,
+		};
+	}, [] );
+
+	const isHomePageBlog = frontPage === postsPage;
+
+	if ( ! isHomePageBlog ) {
+		const homePageIndex = pages?.findIndex(
+			( item ) => item.id === frontPage
+		);
+		const homePage = pages?.splice( homePageIndex, 1 );
+		pages?.splice( 0, 0, ...homePage );
+
+		const postsPageIndex = pages?.findIndex(
+			( item ) => item.id === postsPage
+		);
+
+		const blogPage = pages?.splice( postsPageIndex, 1 );
+
+		pages?.splice( 1, 0, ...blogPage );
+	}
 
 	return (
 		<SidebarNavigationScreen
 			title={ __( 'Pages' ) }
 			description={ __( 'Browse and edit pages on your site.' ) }
 			content={
 				<>
-					{ isLoading && (
+					{ ( isLoadingPages || isLoadingTemplates ) && (
 						<ItemGroup>
 							<Item>{ __( 'Loading pages' ) }</Item>
 						</ItemGroup>
 					) }
-					{ ! isLoading && (
-						<>
-							<SidebarNavigationSubtitle>
-								{ __( 'Recent' ) }
-							</SidebarNavigationSubtitle>
-							<ItemGroup>
-								{ ! pages?.length && (
-									<Item>{ __( 'No page found' ) }</Item>
-								) }
-								{ pages?.map( ( page ) => (
+					{ ! ( isLoadingPages || isLoadingTemplates ) && (
+						<ItemGroup>
+							{ ! pagesAndTemplates?.length && (
+								<Item>{ __( 'No page found' ) }</Item>
+							) }
+							{ isHomePageBlog && homeTemplate && (
+								<PageItem
+									postType="wp_template"
+									postId={ homeTemplate.id }
+									key={ homeTemplate.id }
+									icon={ home }
+									withChevron
+								>
+									{ decodeEntities(
+										homeTemplate.title?.rendered
+									) ?? __( '(no title)' ) }
+								</PageItem>
+							) }
+							{ pages?.map( ( item ) => {
+								let itemIcon;
+								switch ( item.id ) {
+									case frontPage:
+										itemIcon = home;
+										break;
+									case postsPage:
+										itemIcon = loop;
+										break;
+									default:
+										itemIcon = page;
+								}
+								return (
 									<PageItem
-										postId={ page.id }
-										key={ page.id }
+										postId={ item.id }
+										key={ item.id }
+										icon={ itemIcon }
 										withChevron
 									>
 										{ decodeEntities(
-											page.title?.rendered
+											item.title?.rendered
+										) ?? __( '(no title)' ) }
+									</PageItem>
+								);
+							} ) }
+							<VStack className="edit-site-sidebar-navigation-screen__sticky-section">
+								{ dynamicPageTemplates?.map( ( item ) => (
+									<PageItem
+										postType="wp_template"
+										postId={ item.id }
+										key={ item.id }
+										icon={ layout }
+										withChevron
+									>
+										{ decodeEntities(
+											item.title?.rendered
```

Comment:
Can we end up with no title in these templates? 🤔

Reply:
I don't _think_ so because the dynamic templates should always have a name.